course/models.py

```diff
@@ -341,7 +338,10 @@ def get_visible(self, user=None):
         if not user or not user.is_authenticated:
             return self.filter(visible_to_students=True)
         if not user.is_superuser:
-            return self.filter(get_course_visibility_filter(user)).distinct()
+            return self.filter(
+                get_course_visibility_filter(user)
```

Comment:
I think the function name could be updated to reflect the "new" role (i.e. doesn't take care of the visible to student anymore).